app/CurrikiGo/Canvas/Playlist.php

```diff
@@ -47,68 +52,52 @@ public function __construct(Client $client)
      * @param array $data 
      * @return array
      */
-    public function send(PlaylistModel $playlist, $data)
+    public function send(PlaylistModel $playlist, $data, $createAssignment, $canvasCourseId)
     {
-        $user = Auth::user();
-        $projectNameSlug = CourseHelper::urlTitle($playlist->project->name);
-        $sisId = $projectNameSlug . '-' . $user->id . '-' . $playlist->project->id;
-        
-        $lmsSettings = $this->canvasClient->getLmsSettings();
-        $playlistItem = null;
-        $moduleName = Client::CURRIKI_MODULE_NAME;
-        $accountId = "self";
+        try {
+            $user = Auth::user();
+            $projectNameSlug = CourseHelper::urlTitle($playlist->project->name);
+            $sisId = $projectNameSlug . '-' . $user->id . '-' . $playlist->project->id;
 
-        $courses = $this->canvasClient->run(new GetCoursesCommand($accountId, $playlist->project->name, $sisId));
-        $course = CourseHelper::getBySisId($courses, $sisId);
-        
-        if ($course) {
-            // enroll user to existing course as teacher if not enrolled
-            $enrollments = $this->canvasClient->run(new GetCourseEnrollmentCommand($course->id, '?type[]=TeacherEnrollment'));
-            if ($lmsSettings->lms_login_id && !EnrollmentHelper::isEnrolled($lmsSettings->lms_login_id, $enrollments)) {
-                $users = $this->canvasClient->run(new GetUsersCommand($accountId, '?search_term=' . $lmsSettings->lms_login_id));
-                $userIndex = array_search($lmsSettings->lms_login_id, array_column($users, 'login_id'));
-                $user = $userIndex !== false ? $users[$userIndex] : null;
-                if ($user) {
-                    $enrollmentData = ['enrollment' => ['user_id' => $user->id, 'type' => 'TeacherEnrollment', 'enrollment_state' => 'active', 'notify' => true]];
-                    $this->canvasClient->run(new CreateCourseEnrollmentCommand($course->id, $enrollmentData));
-                }
-            }
+            $lmsSettings = $this->canvasClient->getLmsSettings();
+            $playlistItem = null;
+            $moduleName = Client::CURRIKI_MODULE_NAME;
+            $accountId = "self";
+
+            if ($createAssignment == config('constants.canvas_creation_type.create_modules')) {
+                $module = $this->canvasClient->run(new CreateModuleCommand($canvasCourseId, ["name" => $moduleName]));
+                $moduleItem['title'] = $playlist->title . ($data['counter'] > 0 ? ' (' . $data['counter'] . ')' : '');
+                $moduleItem['content_id'] = $playlist->id;
+                $moduleItem['external_url'] = config('constants.curriki-tsugi-host') . "?playlist=" . $playlist->id;
+                $playlistItem = $this->canvasClient->run(new CreateModuleItemCommand($canvasCourseId, $module->id, $moduleItem));
+                $response = 'Modules published successfully';
```

Comment:
You need to first check if the module is not available, and only create if it is.
That snippet is missing.

            // add playlist to existing course
            $modules = $this->canvasClient->run(new GetModulesCommand($course->id, $moduleName));
            $module = CourseHelper::getModuleByName($modules, $moduleName);
            if (!$module) {
                $module = $this->canvasClient->run(new CreateModuleCommand($course->id, ["name" => $moduleName]));
                if ($createAssignmentGroup) {
                    foreach ($activities as $activity) {
                        $createAssignment = $this->canvasClient->run(new CreateAssignmentCommand($canvasCourseId, $createAssignmentGroup->id, $activity->title, $activity->id));
                    }
                }
                $response = 'Assignments published successfully';
            }